data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/accumulator/LocalFileBufferFactory.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.accumulator;
+
+public class LocalFileBufferFactory implements BufferFactory {
+    @Override
+    public Buffer getBuffer() {
+        return new LocalFileBuffer();
```

Comment:
You can probably improve your testing by choosing the `File` here and passing it in as a parameter.

```
File = File.createTempDir(...);
return new LocalFileBuffer(file);
```

Reply:
Addressed.